app/lib/common/providers/common_providers.dart

```diff
@@ -61,50 +60,43 @@ final chatProfileDataProvider =
   return ProfileData(displayName.text(), avatar.data());
 });
 
-final chatsProvider = FutureProvider<List<Convo>>((ref) async {
+final chatsProvider = FutureProvider<List<ffi.Convo>>((ref) async {
   final client = ref.watch(clientProvider)!;
   // FIXME: how to get informed about updates!?!
   final chats = await client.convos();
   return chats.toList();
 });
 
 final chatProvider =
-    FutureProvider.family<Convo, String>((ref, roomIdOrAlias) async {
+    FutureProvider.family<ffi.Convo, String>((ref, roomIdOrAlias) async {
   final client = ref.watch(clientProvider)!;
   // FIXME: fallback to fetching a public data, if not found
   return await client.convo(roomIdOrAlias);
 });
 
 final chatMembersProvider =
-    FutureProvider.family<List<Member>, String>((ref, roomIdOrAlias) async {
+    FutureProvider.family<List<ffi.Member>, String>((ref, roomIdOrAlias) async {
   final chat = await ref.watch(chatProvider(roomIdOrAlias).future);
   final members = await chat.activeMembers();
   return members.toList();
 });
 
 final relatedChatsProvider = FutureProvider.autoDispose
-    .family<List<Convo>, String>((ref, spaceId) async {
+    .family<List<ffi.Convo>, String>((ref, spaceId) async {
   return (await ref.watch(spaceRelationsOverviewProvider(spaceId).future))
       .knownChats;
 });
 
 // Member Providers
 final memberProfileProvider =
-    FutureProvider.family<ProfileData, String>((ref, userId) async {
-  final member = ref.watch(memberProvider(userId));
-  return member.maybeWhen(
-    data: (data) async {
-      UserProfile profile = data!.getProfile();
-      OptionString displayName = await profile.getDisplayName();
-      final avatar = await profile.getThumbnail(62, 60);
-      return ProfileData(displayName.text(), avatar.data());
-    },
-    orElse: () => ProfileData('', null),
-  );
-});
-
-final memberProvider =
-    FutureProvider.family<Member?, String>((ref, userId) async {
-  final convo = ref.watch(currentConvoProvider);
-  return await convo!.getMember(userId);
+    FutureProvider.family<ProfileData, ffi.Member>((ref, member) async {
+  try {
+    ffi.UserProfile profile = member.getProfile();
+    ffi.OptionString displayName = await profile.getDisplayName();
+    final avatar = await profile.getThumbnail(62, 60);
+    return ProfileData(displayName.text(), avatar.data());
+  } catch (e) {
+    debugPrint('$e');
+    return ProfileData('', null);
+  }
```

Comment:
I think we should find some agreed upon patterns and write them up, because some of these constrains are really detrimental for the user experience and we need to make sure those don't fall through the cracks.